File: Python/Arrays/L_wordSearch.py
def exist(board, word):
    m,n = len(board), len(board[0])
    path = set()
                
    def dfs(i,j,l):
        if len(word) == l:
            return True
        if (i < 0 or j < 0 or i >= m or j >= n or word[l] != board[i][j] or (i,j) in path):
            return False
        path.add((i,j))
        res = (dfs(i+1, j, l+1) or dfs(i, j+1, l+1) or dfs(i-1, j, l+1) or dfs(i, j-1, l+1))
        path.remove((i,j))
        return res
        
    for i in range(m):
        for j in range(n):
            if dfs(i,j,0): return True
    return False

File: cpp/Backtracking/L_wordSearch.cpp
#include<bits/stdc++.h>
using namespace std;


bool exist(vector<vector<char>>& board, string word) {
    int r = board.size(), c = board[0].size();
    for(int i=0;i<r;i++) {
        for(int j=0;j<c;j++) {
            if(board[i][j] == word[0] and dfs(i,j,r,c,0,board,word)) return true;
        }
    }
    return false;
}
bool dfs(int i, int j, int r, int c, int idx, vector<vector<char>>& board, string word) {
    if(i < 0 or i >= r or j < 0 or j >= c or word[idx] != board[i][j]) return false;
    if(idx == word.size() - 1) return true;
    board[i][j] = '-';
    if(dfs(i+1,j,r,c,idx+1,board,word) or
        dfs(i-1,j,r,c,idx+1,board,word) or
        dfs(i,j+1,r,c,idx+1,board,word) or
        dfs(i,j-1,r,c,idx+1,board,word)) return true;
    board[i][j] = word[idx];
    return false;
}

int main() {

}